src/Command.js

```diff
@@ -17,45 +17,69 @@ export default class Command {
   }
 
   run() {
+    this.logger.info("Lerna v" + this.lernaVersion);
+
+    if (this.repository.isIndependent()) {
+      this.logger.info("Independent Versioning Mode");
+    }
+
+    this.runValidations();
     this.runPreparations();
     this.runCommand();
   }
 
-  runPreparations() {
-    this.logger.info("Lerna v" + this.lernaVersion);
+  runValidations() {
+    if (!FileSystemUtilities.existsSync(this.repository.packagesLocation)) {
+      this.logger.warning("`packages/` directory does not exist, have you run `lerna init`?");
+      this._complete(null, 1);
+      return;
+    }
 
-    if (this.flags.independent) {
-      this.logger.info("Independent Versioning Mode");
+    if (!FileSystemUtilities.existsSync(this.repository.packageJsonLocation)) {
+      this.logger.warning("`package.json` does not exist, have you run `lerna init`?");
+      this._complete(null, 1);
+      return;
     }
 
-    if (this.flags.canary) {
-      this.logger.info("Publishing canary build");
+    if (!FileSystemUtilities.existsSync(this.repository.lernaJsonLocation)) {
+      this.logger.warning("`lerna.json` does not exist, have you run `lerna init`?");
+      this._complete(null, 1);
+      return;
     }
 
-    if (!FileSystemUtilities.existsSync(this.repository.packagesLocation)) {
-      this.logger.error("`packages/` directory does not exist, have you run `lerna init`?");
-      this._complete(1);
+    if (this.flags.independent && !this.repository.isIndependent()) {
+      this.logger.warning("You ran lerna with `--independent` or `-i`, but the repository is not set to independent mode");
```

Comment:
Possibly explain that "setting the repo to independent mode" means adding the `version` key to `independent`